CONTRIBUTING.rst

```diff
@@ -0,0 +1,678 @@
+    using the symbolic name of the error as reported by pylint.
+
+.. code-block:: python
+
+    import airflow.* # pylint: disable=wildcard-import
+
+
+-   If there is a single line where you need to disable a particular error,
+    consider adding a comment to the line that causes the problem. For example:
+
+.. code-block:: python
+
+    def  MakeSummary(pcoll, metric_fn, metric_keys): # pylint: disable=invalid-name
+
+
+-   For multiple lines/block of code, to disable an error, you can surround the
+    block with ``pylint:disable/pylint:enable`` comment lines. For example:
+
+.. code-block:: python
+
+    # pylint: disable=too-few-public-methods
+    class  LoginForm(Form):
+    """Form for the user"""
+    username = StringField('Username', [InputRequired()])
+    password = PasswordField('Password', [InputRequired()])
+    # pylint: enable=too-few-public-methods
+
+
+Pre-commit Hooks
+================
+
+Pre-commit hooks are a fantastic way of speeding up your local development
+cycle, either in local virtualenv or Breeze.
+
+The pre-commit hooks only check the files you are currently working on and make
+them fast. Yet, these checks use exactly the same environment as the CI tests
+use. So, you can be sure your modifications will also work for CI if they pass
+pre-commit hooks.
+
+You are **STRONGLY** encouraged to install the pre-commit hooks as they speed up
+your development and place less burden on the CI infrastructure.
+
+We have integrated the fantastic `pre-commit <https://pre-commit.com>`__. framework
+in our development workflow. To install and use it, you need Python 3.6 locally.
+
+It is the best to use pre-commit hooks when you have your local virtualenv for
+Airflow activated since then pre-commit hooks and other dependencies are
+automatically installed. You can also install the pre-commit hooks manually
+using ``pip install``.
+
+The pre-commit hooks require the Docker Engine to be configured as the static
+checks are executed in the Docker environment. You should build the images
+locally before installing pre-commit checks as described in [Breeze user
+guide](BREEZE.rst). In case you do not have your local images built, the
+pre-commit hooks fail and provide instructions on what needs to be done.
+//didn't get this. above you refer to local virtualenv but now say it's Docker,
+i.e. Breeze?
+
+Prerequisites for Pre-commit Hooks
+----------------------------------
+
+The pre-commit hooks use several external linters that need to be installed
+before pre-commit is run.
+
+Each of the checks installs its own environment, so you do not need to install
+those, but there are some
+
+checks that require locally installed binaries. On Linux, you typically install
+them with ``sudo apt install``, on macOS - with ``brew install``.
+
+The current list of prerequisites:
+
+-   ``xmllint``: on Linux, install via ``sudo apt install xmllint``; on macOS,
+    install via ``brew install xmllint``
+
+Enabling Pre-commit Hooks
```

Comment:
Elena: my concern is that we say above that pre-commit hooks are installed automatically as part of local virtualenv, or manually with pip. Probably here we're talking about enabling?
Jarek: fine for me. The command is `install` but we can use "enable" as name.